test_simpleeval.py

```diff
@@ -1351,5 +1352,29 @@ def test_no_operators(self):
             s.eval("~ 2")
 
 
+class TestAllowedAttributes(DRYTest):
+    def test_allowed_attrs_(self):
+        self.s.allowed_attrs = BASIC_ALLOWED_ATTRS
+        self.t("5 + 5", 10)
+        self.t('"   hello  ".strip()', "hello")
+
+    def test_breakout_via_generator(self):
+        # Thanks decorator-factory
+        class Foo:
+            def bar(self):
+                yield "Hello, world!"
+
+        # Test the genertor does work - also adds the `yield` to codecov...
+        assert list(Foo().bar()) == ["Hello, world!"]
+
+        evil = "foo.bar().gi_frame.f_globals['__builtins__'].exec('raise RuntimeError(\"Oh no\")')"
+
+        x = simpleeval.DISALLOW_METHODS
+        simpleeval.DISALLOW_METHODS = []
+        with self.assertRaises(FeatureNotAvailable):
```

Comment:
It will be valuable to confirm which `FeatureNotAvailable` path is getting trigger here.

Using pytest, this would be something like `with pytest.raises(FeatureNotAvailable, match=r"'.+\.gi_frame' access not allowed"`. I don't know if Python's unittest supports this style of invocation, but it'll be good to verify the exception's message matches expectations.

Reply:
Would it be helpful to subclass `FeatureNotAvailable` do you think?

(Not just for the tests, but for general library use?)

Reply:
I don't know whether it would helpful to users, but I recommend checking the message in the exception text.

I found that `self.assertRaisesRegex()` does something similar to `pytest.raises()` [[docs link](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex)]:

```python
self.assertRaisesRegex(FeatureNotAvailable, r"'.+\.gi_frame` access not allowed"):
    ...
```